samples/snippets/publisher.py

```diff
@@ -273,6 +273,82 @@ def publish_messages_with_retry_settings(project_id, topic_id):
     # [END pubsub_publisher_retry_settings]
 
 
+def publish_with_ordering_keys(project_id, topic_id):
+    """Publishes messages with ordering keys."""
+    # [START pubsub_publish_with_ordering_keys]
+    from google.cloud import pubsub_v1
+
+    # TODO(developer): Choose an existing topic.
+    # project_id = "your-project-id"
+    # topic_id = "your-topic-id"
+
+    publisher_options = pubsub_v1.types.PublisherOptions(
+        enable_message_ordering=True
+    )
+    publisher = pubsub_v1.PublisherClient(publisher_options=publisher_options)
+    # The `topic_path` method creates a fully qualified identifier
+    # in the form `projects/{project_id}/topics/{topic_id}`
+    topic_path = publisher.topic_path(project_id, topic_id)
+
+    for message in [
+        ("message1", "key1"),
+        ("message2", "key2"),
+        ("message3", "key1"),
+        ("message4", "key2"),
+    ]:
+        # Data must be a bytestring
+        data = message[0].encode("utf-8")
+        ordering_key = message[1]
+        # When you publish a message, the client returns a future.
+        future = publisher.publish(
+            topic_path, data=data, ordering_key=ordering_key
+        )
+        print(future.result())
+
+    print("Published messages with ordering keys.")
+    # [END pubsub_publish_with_ordering_keys]
+
+
+def resume_publish_with_ordering_keys(project_id, topic_id):
+    """Resume publishing messages with ordering keys when unrecoverable errors occur."""
+    # [START pubsub_resume_publish_with_ordering_keys]
+    from google.cloud import pubsub_v1
+
+    # TODO(developer): Choose an existing topic.
+    # project_id = "your-project-id"
+    # topic_id = "your-topic-id"
+
+    publisher_options = pubsub_v1.types.PublisherOptions(
+        enable_message_ordering=True
```

Comment:
Likewise here for the regional endpoint.

Reply:
Done.